### PR TITLE
[opencl-aot] Support more available CPUs for march option.

### PR DESCRIPTION
**Pull request by haonanya**

### CODE REVIEW

File: opencl/opencl-aot/source/main.cpp [hunk 1]
```diff
@@ -322,10 +322,29 @@ int main(int Argc, char *Argv[]) {
...
+    skx,
+    clk,
+    icl,
+    icx,
+    spr
+  };
   cl::opt<ArchType> OptMArch(
-      "march",
-      cl::desc("Set target CPU architecture (for --device=cpu or "
+      "March",
```
Comment (haonanya) [inline opencl/opencl-aot/source/main.cpp:344]: march is used in LLVM CodeGen https://github.com/intel/llvm/blob/sycl/llvm/lib/CodeGen/CommandFlags.cpp#L115.

File: opencl/opencl-aot/source/main.cpp [hunk 2]
```diff
@@ -477,7 +504,19 @@ int main(int Argc, char *Argv[]) {
...
+        {avx512, "skx"},
+        {wsm, "corei7"},
+        {snb, "corei7-avx"},
+        {ivyb, "corei7-avx"},
+        {bdw, "core-avx2"},
+        {cfl, "core-avx2"},
+        {adl, "core-avx2"},
+        {skylake, "core-avx2"},
+        {skx, "skx"},
+        {clk, "skx"},
```
Comment (hewj03) [inline opencl/opencl-aot/source/main.cpp:516]:
```suggestion
        {clk, "cascadelake"},
```

File: opencl/opencl-aot/source/main.cpp (hunk 1 above)
Comment (asudarsa) [reply opencl/opencl-aot/source/main.cpp:344]: Hi @haonanya 

I have similar concerns. Can you please provide a tool flow where this tool is built in conjunction with LLVM codegen? Also, I see both the options have existed for atleast three years now. I am curious to see why did the issue crop up just now?

Thanks

Comment (haonanya) [reply opencl/opencl-aot/source/main.cpp:344]: Thanks, I will continue to investigate it deeper.

File: opencl/opencl-aot/source/main.cpp [hunk 3]
```diff
@@ -347,12 +366,20 @@ int main(int Argc, char *Argv[]) {
...
-          clEnumValN(
-              sse42, "sse4.2",
+          clEnumVal(
+              sse42,
               "Enable support of Intel(R) SSE4.2 Efficient Accelerated String "
               "and Text Processing Instructions, Intel(R) SSE4 Vectorizing "
               "Compiler and Media Accelerator, Intel(R) SSE3, Intel(R) SSE2, "
-              "Intel(R) SSE, and SSSE3 instructions")));
+              "Intel(R) SSE, and SSSE3 instructions"),
+          clEnumVal(wsm, "Westmere"), clEnumVal(snb, "Sandybridge"),
```
[the remark this answers is not in the document]
Comment (haonanya) [reply opencl/opencl-aot/source/main.cpp]: Thanks, @dm-vodopyanov , updated, please take a look.

File: opencl/opencl-aot/source/main.cpp [hunk 4]
```diff
@@ -322,37 +322,36 @@ int main(int Argc, char *Argv[]) {
                         cl::desc("Input OpenCL program binary files"));
   cl::alias OptIR("ir", cl::aliasopt(OptOutputElf), cl::desc("Output file"));
 
-  enum ArchType : int8_t { sse42 = 0, avx, avx2, avx512 };
+  enum ArchType : int8_t {
+    wsm,
```
Comment (dm-vodopyanov) [inline opencl/opencl-aot/source/main.cpp:326]: I'm pretty sure these new options will ruin customers applications. Customers use old values of `-march` option, and it's too tough to remove them in the single release.
For the current release, can you keep the old values, link them to the new architectures and print a warning highlighting that these old values will be removed in the next release and suggest how users should replace old values with the new ones? And in the next release you can remove these old options completely.

File: opencl/opencl-aot/source/main.cpp [hunk 5]
```diff
@@ -322,37 +322,36 @@ int main(int Argc, char *Argv[]) {
...
-              "and Text Processing Instructions, Intel(R) SSE4 Vectorizing "
-              "Compiler and Media Accelerator, Intel(R) SSE3, Intel(R) SSE2, "
-              "Intel(R) SSE, and SSSE3 instructions")));
+          clEnumVal(wsm, "Intel® microarchitecture code name Westmere"),
+          clEnumVal(snb, "Intel® microarchitecture code name Sandy Bridge"),
+          clEnumVal(ivyb, "Intel® microarchitecture code name Ivy Bridge"),
+          clEnumVal(bdw, "Intel® microarchitecture code name Broadwell"),
+          clEnumVal(cfl, "Coffee Lake"), clEnumVal(adl, "Alder Lake"),
+          clEnumVal(skylake, "Intel® microarchitecture code name Skylake"),
+          clEnumVal(skx, "Intel® microarchitecture code name Skylake"),
```
Comment (dm-vodopyanov) [inline opencl/opencl-aot/source/main.cpp:351]: Same description for two different values

File: opencl/opencl-aot/source/main.cpp [hunk 6]
```diff
@@ -322,37 +322,36 @@ int main(int Argc, char *Argv[]) {
...
+          clEnumVal(clk, "Cascade Lake"), clEnumVal(icl, "Ice Lake"),
+          clEnumVal(icx, "Ice Lake"), clEnumVal(spr, "Sapphire Rapids"),
```
Comment (dm-vodopyanov) [inline opencl/opencl-aot/source/main.cpp:353]: The same for ICL and ICX

File: opencl/opencl-aot/source/main.cpp (hunk 4 above)
Comment (haonanya) [reply opencl/opencl-aot/source/main.cpp:330]: Thanks! Restore old options. This PR will only add more options for march. Removal of old options maybe need further discussion and could be done on follow-up patch if the idea is approved.

File: opencl/opencl-aot/source/main.cpp (hunk 2 above)
Comment (dm-vodopyanov) [reply opencl/opencl-aot/source/main.cpp:519]: Unresolving conversation, it wasn't fixed

**Review by dm-vodopyanov — APPROVED** [review]

**Review by wenju-he — APPROVED** [review]
LGTM

**Review by cdai2 — APPROVED** [review]

**Comment by haonanya** [comment]
@intel/dpcpp-tools-reviewers , do you have any comments? Thanks.

**Pull request merged**